soroban-env-common/src/val.rs

```diff
@@ -820,8 +817,12 @@ impl Debug for Val {
             Tag::SymbolSmall => {
                 let ss: SymbolStr =
                     unsafe { <SymbolSmall as ValConvert>::unchecked_from_val(*self) }.into();
-                let s: &str = ss.as_ref();
-                write!(f, "Symbol({s})")
+                // Even though this may be called for an arbitrary, not necessarily well-formed
+                // `Val`, this is still safe thanks to `SymbolSmall` iteration implementation that
+                // only returns valid symbol characters or `\0` even for invalid bit
+                // representations.
+                let converted_str = unsafe { str::from_utf8_unchecked(ss.as_ref()) };
+                write!(f, "Symbol({})", converted_str)
```

Comment:
Is the introduction of the unsafe unchecked required here? Why can't we keep using the `let s: &str = ss.as_ref()`? I must be missing something because this is the second spot that looks like an `AsRef<&str>` might have been removed, but I can't see it being removed anywhere in this PR.

Reply:
Again, I still think we can remove it, just need to agree upon this. For now, I wanted to obviate the unsafe conversion usage (this is the only code path that actually requires is now). If we decide that it's important to keep `as_ref`, I'll revert this.

Reply:
I reverted the changes related to `as_ref` for now.